### PR TITLE
[new release] websocket, websocket-lwt-unix and websocket-async (2.13)

### PR DESCRIPTION
**Pull request by vbmithr**
Websocket library

- Project page: <a href="https://github.com/vbmithr/ocaml-websocket">https://github.com/vbmithr/ocaml-websocket</a>
- Documentation: <a href="https://vbmithr.github.io/ocaml-websocket/doc">https://vbmithr.github.io/ocaml-websocket/doc</a>

##### CHANGES:

- Async: use logs_async
- Async: upgrade_connection API (@kkazuo)
- Lwt: fix upgrade_connection for TLS (API change)
- Upgrade Base64 support to 3.1.0

### CODE REVIEW

File: packages/websocket-lwt-unix/websocket-lwt-unix.2.13/opam [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+maintainer: "Vincent Bernardoff <vb@luminar.eu.org>"
+homepage: "https://github.com/vbmithr/ocaml-websocket"
+bug-reports: "https://github.com/vbmithr/ocaml-websocket/issues"
+dev-repo: "git+https://github.com/vbmithr/ocaml-websocket"
+doc: "https://vbmithr.github.io/ocaml-websocket/doc"
+tags: [
+  "org:mirage"
+  "org:xapi-project"
+]
+build: [ "dune" "build" "-j" jobs "-p" name "@install" ]
```
Comment (kit-ty-kate) [inline packages/websocket-lwt-unix/websocket-lwt-unix.2.13/opam]: This is not a blocker at all but just for the information `"@install"` is not necessary

File: packages/websocket-async/websocket-async.2.13/opam [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+dev-repo: "git+https://github.com/vbmithr/ocaml-websocket"
+doc: "https://vbmithr.github.io/ocaml-websocket/doc"
+tags: [
+  "org:mirage"
+  "org:xapi-project"
+]
+build: [ "dune" "build" "-j" jobs "-p" name "@install" ]
+depends: [
+  "dune" {build & >= "1.1.1"}
+  "websocket" {= "2.12"}
```
Comment (kit-ty-kate) [inline packages/websocket-async/websocket-async.2.13/opam]: this doesn't seem to be the right version number

**Comment by kit-ty-kate** [comment]
`websocket-async.2.13` fails with:
```
#=== ERROR while compiling websocket-async.2.13 ===============================#
# context              2.0.3 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/vbmithr/ocaml-websocket/releases/download/2.13/websocket-2.13.tbz)
# path                 ~/.opam/4.07/.opam-switch/build/websocket-async.2.13
# command              ~/.opam/4.07/bin/dune build -j 71 -p websocket-async @install
# exit-code            1
# env-file             ~/.opam/log/websocket-async-20-d2e822.env
# output-file          ~/.opam/log/websocket-async-20-d2e822.out
### output ###
# File "async/dune", line 13, characters 25-44:
# 13 |  (libraries uri.services logs-async-reporter websocket-async))
#                               ^^^^^^^^^^^^^^^^^^^
# Error: Library "logs-async-reporter" not found.
# Hint: try: dune external-lib-deps --missing -j 71 -p websocket-async @install
```
And `websocket-lwt.2.13` is also missing from this PR:
```
$ opam depext -uivyj 2 websocket-lwt-unix.2.13
# Detecting depexts using vars: arch=x86_64, os=linux, os-distribution=debian, os-family=debian
[ERROR] No solution for websocket-lwt-unix.2.13: The following dependencies couldn't be met:
          - websocket-lwt-unix -> websocket-lwt >= 2.13
              no matching version
```

File: packages/websocket-async/websocket-async.2.13/opam (hunk 2 above)
Comment (avsm) [reply packages/websocket-async/websocket-async.2.13/opam]: you can now use `{=version}` to make the constraint track the current package version

**Comment by kit-ty-kate** [comment]
There is an issue on 4.04 and 4.05:
```
#=== ERROR while compiling websocket-lwt-unix.2.13 ============================#
# context              2.0.3 | linux/x86_64 | ocaml-base-compiler.4.04.2 | pinned(https://github.com/vbmithr/ocaml-websocket/releases/download/2.13/websocket-2.13.tbz)
# path                 ~/.opam/4.04/.opam-switch/build/websocket-lwt-unix.2.13
# command              ~/.opam/4.04/bin/dune build -j 71 -p websocket-lwt-unix
# exit-code            1
# env-file             ~/.opam/log/websocket-lwt-unix-20-3730ca.env
# output-file          ~/.opam/log/websocket-lwt-unix-20-3730ca.out
### output ###
#     ocamldep lwt/.websocket_lwt_unix.objs/websocket_lwt_unix.mli.d (exit 2)
# (cd _build/default && /home/opam/.opam/4.04/bin/ocamldep.opt -modules -intf lwt/websocket_lwt_unix.mli) > _build/default/lwt/.websocket_lwt_unix.objs/websocket_lwt_unix.mli.d
# File "lwt/websocket_lwt_unix.mli", line 25, characters 20-22:
# Error: Syntax error
```

**Comment by kit-ty-kate** [comment]
ping @vbmithr

**Comment by kit-ty-kate** [comment]
Could you have a look at the revdeps failures for `disml` and `gimini`? Those two were not failing before

**Comment by mseri** [comment]
@kit-ty-kate disml is fixed no, gimini is still failing with some async-related error. @vbmithr, do you know if this is related to this PR? Maybe it can be enough to add an upper bound to gimini opam files.

**Comment by vbmithr** [comment]
Yes gemini needs an upper bound…

**Comment by mseri** [comment]
Would you mind adding it to this PR? Otherwise we could merge and add it separately

**Comment by vbmithr** [comment]
Done… hope this is fine this time…

**Comment by kit-ty-kate** [comment]
Thanks a lot!

**Pull request merged**